core/src/main/java/org/apache/carbondata/core/scan/model/QueryModel.java

```diff
@@ -133,7 +134,11 @@ public static void processFilterExpression(FilterProcessVO processVO, Expression
       final boolean[] isFilterDimensions, final boolean[] isFilterMeasures,
       CarbonTable carbonTable) {
     if (null != filterExpression) {
-      if (null != filterExpression.getChildren() && filterExpression.getChildren().size() == 0) {
+      if (filterExpression instanceof PolygonExpression) {
+        /* Process polygon expression */
+        ((PolygonExpression) filterExpression).buildRangeExpression();
```

Comment:
can buildRangeExpression during the constructor itself ? Better to do as soon as object is created.

Reply:
Have deferred the range expresssions build until the point required. Because, there can be many range expressions. If we build complete expression tree early, we will have huge data to serialize and deserialize. Thought it degrades the performance.